ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -287,21 +288,14 @@ public void deleteInternalTopics(final String applicationId) {
         deleteTopics(internalTopics);
       }
     } catch (final Exception e) {
-      LOG.error("Exception while trying to clean up internal topics for application id: {}.",
-          applicationId, e
-      );
-    }
-  }
-
-  private boolean isTopicDeleteEnabled() {
-    try {
-      final ConfigEntry configEntry = getConfig().get(DELETE_TOPIC_ENABLE);
-      // default to true if there is no entry
-      return configEntry == null || Boolean.valueOf(configEntry.value());
-    } catch (final Exception e) {
-      LOG.error("Failed to initialize TopicClient: {}", e.getMessage());
-      throw new KafkaResponseGetFailedException(
-          "Could not fetch broker information. KSQL cannot initialize", e);
+      // An exception due to TopicDeletionDisabledException should not be logged as an error, just
+      // info level should be enough. However, the deleteTopics() method already logged it for us,
+      // so we can skip it.
+      if (!(e instanceof TopicDeletionDisabledException)) {
```

Comment:
nasty! We can clean this up with we don't log and throw, but just throw in `deleteTopics`.

Reply:
Done